src/Context/Paths.hs

```diff
@@ -0,0 +1,39 @@
+module Context.Paths where
+
+import Base
+import Context.Type
+import Hadrian.Expression
+
+-- | The directory to the current stage
+stageDir :: Context -> FilePath
+stageDir Context {..} = stageString stage
+
+-- | The path to the current stage
+stagePath :: Context -> Action FilePath
+stagePath context = buildRoot <&> (-/- stageDir context)
+
+getStagePath :: Expr Context b FilePath
+getStagePath = expr . stagePath =<< getContext
+
+-- | The directory in 'buildRoot' containing build artifacts of a given 'Context'.
+contextDir :: Context -> FilePath
+contextDir Context {..} = stageString stage -/- pkgPath package
+
+-- | Path to the context directory, containing the "build folder"
+contextPath :: Context -> Action FilePath
+contextPath context = buildRoot <&> (-/- contextDir context)
+
+getContextPath :: Expr Context b FilePath
+getContextPath = expr . contextPath =<< getContext
+
+-- | The directory in 'buildRoot' containing the object artifacts.
+buildDir :: Context -> FilePath
```

Comment:
Can you add a clarification of the difference between `contextDir` and `buildDir`? Why do we need both? Can we get rid of `buildDir`?

Reply:
I don't remember if this is some left over from multiple refactorings or if `cabal` had some expectation of `build`. Might very well be from the former.